sparkmagic/sparkmagic/tests/test_configuration.py

```diff
@@ -78,3 +78,10 @@ def test_share_config_between_pyspark_and_pyspark3():
     kpc = { 'username': 'U', 'password': 'P', 'base64_password': 'cGFzc3dvcmQ=', 'url': 'L', 'auth': AUTH_BASIC }
     overrides = { conf.kernel_python_credentials.__name__: kpc }
     assert_equals(conf.base64_kernel_python3_credentials(), conf.base64_kernel_python_credentials())
+
+@with_setup(_setup)
+def test_custom_headers():
```

Comment:
Sorry if I was not clear, I meant that you should add a unit test to test that the from_endpoint method in LivyReliableHttpClient returns a client with custom headers if some custom headers are specified in the config.